src/mono/wasm/runtime/driver.c

```diff
@@ -1132,3 +1123,26 @@ mono_wasm_intern_string (MonoString *string)
 {
 	return mono_string_intern (string);
 }
+
+EMSCRIPTEN_KEEPALIVE void
+mono_wasm_string_get_data (
+	MonoString *string, mono_unichar2 **outChars, int *outLengthBytes, int *outIsInterned
+) {
+	if (!string) {
+		if (outChars)
+			*outChars = 0;
+		if (outLengthBytes)
+			*outLengthBytes = 0;
+		if (outIsInterned)
+			*outIsInterned = 1;
+		return;
+	}
+
+	if (outChars)
+		*outChars = mono_string_chars (string);
+	if (outLengthBytes)
+		*outLengthBytes = mono_string_length (string) * 2;
```

Comment:
Could there be 4 byte character in UTF16 ?

Reply:
yes, but everything in .net and js is utf16, so string_length * 2 is the length in bytes. you have to handle surrogate pairs by hand in both environments